web/src/main/java/cz/muni/fi/pa165/mvc/controllers/UserController.java

```diff
@@ -0,0 +1,163 @@
+
+    @RequestMapping(value = "/{id}/delete", method = RequestMethod.POST)
+    public String delete(@PathVariable long id, UriComponentsBuilder uriBuilder,
+                         RedirectAttributes redirectAttributes) {
+        UserDTO user = userFacade.getUserById(id);
+        try {
+            userFacade.removeUser(id);
+            redirectAttributes.addFlashAttribute("alert_success", "User \"" + user.getLastName() + "\" was deleted.");
+        } catch (Exception ex) {
+            redirectAttributes.addFlashAttribute("alert_danger",
+                    "User \"" + user.getLastName() + "\" cannot be deleted, it might be used for a user.");
+        }
+        return "redirect:" + uriBuilder.path("/user/").toUriString();
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    public String view(@PathVariable long id, Model model) {
+        model.addAttribute("user", userFacade.getUserById(id));
+        model.addAttribute("role", roleFacade.getRoleById(userFacade.getUserById(id).getRole().getId()));
+        return "user/view";
+    }
+
+    @RequestMapping(value = "/{id}/delete", method = RequestMethod.GET)
+    public String confirmDelete(@PathVariable long id, Model model) {
+        model.addAttribute("user", userFacade.getUserById(id));
+        return "user/confirm_delete";
+    }
+
+    /**
+     * Prepares an empty form.
+     *
+     * @param model data to be displayed
+     * @return JSP page
+     */
+    @RequestMapping(value = "/new", method = RequestMethod.GET)
+    public String newUser(Model model) {
+        model.addAttribute("userCreate", new CreateUserDTO());
+        return "user/new";
+    }
+
+    @RequestMapping(value = "/new", method = RequestMethod.POST)
+    public String create(@Valid @ModelAttribute("userCreate") CreateUserDTO formBean, BindingResult bindingResult,
+                         Model model, RedirectAttributes redirectAttributes, UriComponentsBuilder uriBuilder) {
+        if (bindingResult.hasErrors()) {
+            for (FieldError fe : bindingResult.getFieldErrors()) {
+                model.addAttribute(fe.getField() + "_error", true);
+            }
+            return "user/new";
+        }
+
+//       Role role = EntityUtils.createRole(null, "User",  "User of system");
+//
+//        RoleDTO rolee =roleFacade.getRoleByName("User");
+//        //roleService.create(rolee);
+//        CreateRoleDTO roleee=new CreateRoleDTO();
+//        roleee.setDescription("looser");
+//        roleee.setName("Robo");
+//        Long idd =roleFacade.createRole(roleee);
+//
+//
+//        CreateUserDTO user = new CreateUserDTO();
+//        user.setFirstName("Robo");
+//        user.setLastName("Dudas");
+//        user.setPassword("robo123");
+//        user.setEmail("robo@robo.com");
+//        user.setRoleId(rolee.getId());
+//        Long id = userFacade.createUser(user);
+
+        RoleDTO role =roleFacade.getRoleByName("User");
+        formBean.setRoleId(role.getId());
```

Comment:
Kdybys tak tohle věděl... Zkoušel jsi ten přístup, který ve vzorovým příkladu použili při vytváření produktu? Ten by fungovat měl...

Reply:
Tohle nedava zadny smysl. Admin muze vytvorit dalsiho admina, takze zadne hardcodovani toho, ze to je user nedelej, ale nechej zadat ve view